lib/src/geometry/line.dart

```diff
@@ -0,0 +1,44 @@
+import 'dart:math';
+
+import '../../extensions.dart';
+
+/// This represents a linear function on the ax + by = c form
+class Line {
+  final double a;
+  final double b;
+  final double c;
+
+  const Line(this.a, this.b, this.c);
+
+  static Line fromPoints(Vector2 p1, Vector2 p2) {
+    final a = p2.y - p1.y;
+    final b = p1.x - p2.x;
+    final c = p2.y * p1.x - p1.y * p2.x;
+    return Line(a, b, c);
+  }
+
+  /// Returns an empty list if there is no intersection
```

Comment:
I probably wrote this but would be good to add:

* If they are concurrent returns exactly 1 point.
* If they coincide returns empty list as well (very important to note!)